### PR TITLE
Improve client methods

### PR DESCRIPTION
**Pull request by ehmicky**
**- Summary**

This improves the OpenAPI methods generation. This PR refactors the related code and splits it into several files and functions as opposed to current single big function. It also brings the following improvements:
  - the `body` parameter can be a function generation. However this was not working when using JSON request bodies.
  - the inverse was true as well: the `body` could only be binary when wrapped inside a function. 
  - this removes some dependencies on core Node.js APIs to improve our goal to make this client browser-friendly.
  - the `this` context is now bound, which means users won't run into errors when destructuring (`const { 
getCurrentUser } = api`) or passing the function as argument (`func(api.getCurrentUser)`)
  - the methods now have the proper `function.name` (e.g. `getCurrentUser`) which provides with better debugging (e.g. in stack traces)
  - improved error messages. For example `Missing required params` was not clear whether the parameter was in the path, the body or the query variables. 

**- Test plan**

The current unit tests are passing. We have 100% test coverage on this part of the code.

**- Description for the changelog**

Improves the OpenAPI methods generation.

**- A picture of a cute animal (not mandatory but encouraged)**

penguin-2

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,13 +1,16 @@
 const set = require('lodash.set')
 const get = require('lodash.get')
 const dfn = require('@netlify/open-api')
-const { methods, generateMethod } = require('./open-api')
+const { addMethods } = require('./methods')
+const methods = require('./open-api/shape-swagger')
```
Comment (DavidWells) [inline src/index.js]: Is `methods` returning the same shape?

This is used by the `netlify api` CLI command https://github.com/netlify/cli/blob/master/src/commands/api.js#L5

Comment (ehmicky) [reply src/index.js]: `methods` is unchanged.

There is a separate PR (#66) which also does not change it. I ran `diff` before and after that PR to ensure shape was the same.

Comment (DavidWells) [reply src/index.js]: ok cool 😎

Comment (ehmicky) [reply src/index.js]: I've opened an issue at #75.

**Pull request merged**